src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest_rescheduleCourtHearing.java

```diff
@@ -0,0 +1,116 @@
+                request,
+                ErrorResponse.class);
+
+        assertThat(response.getBody()).isEqualTo(
+                ErrorResponse.builder()
+                        .status(404)
+                        .userMessage("Offender booking with id 9999999 not found.")
+                        .developerMessage("Offender booking with id 9999999 not found.")
+                        .build());
+    }
+
+    @Test
+    public void reschedule_court_hearing_fails_when_date_in_past() {
+        final var token = authTokenHelper.getToken(AuthTokenHelper.AuthToken.NORMAL_USER);
+
+        final var request = createHttpEntity(token, Map.of());
+
+        final var response = testRestTemplate.exchange(
+                "/api/bookings/-8/prison-to-court-hearings/-208/revised-hearing-date/1970-01-01T00:00",
+                HttpMethod.PUT,
+                request,
+                ErrorResponse.class);
+
+        assertThat(response.getBody()).isEqualTo(
+                ErrorResponse.builder()
+                        .status(400)
+                        .userMessage("Revised court hearing date '1970-01-01T00:00' must be in the future.")
+                        .developerMessage("Revised court hearing date '1970-01-01T00:00' must be in the future.")
+                        .build());
+    }
+
+    @Test
+    public void schedules_court_hearing_fails_when_unauthorised() {
+        final var token = authTokenHelper.getToken(AuthTokenHelper.AuthToken.RENEGADE_USER);
+
+        final var request = createHttpEntity(token, Map.of());
+
+        final var response = testRestTemplate.exchange(
+                "/api/bookings/9999999/prison-to-court-hearings/-1/revised-hearing-date/2030-03-11T14:00",
+                HttpMethod.PUT,
+                request,
+                ErrorResponse.class);
+
+        assertThat(response.getBody()).isEqualTo(
+                ErrorResponse.builder()
+                        .status(403)
+                        .userMessage("Access is denied")
+                        .build());
+    }
+
+    @Test
+    public void reschedule_court_hearing_fails_when_hearing_in_past() {
+        final var token = authTokenHelper.getToken(AuthTokenHelper.AuthToken.NORMAL_USER);
+
+        final var request = createHttpEntity(token, Map.of());
+
+        final var response = testRestTemplate.exchange(
+                "/api/bookings/-8/prison-to-court-hearings/-209/revised-hearing-date/2030-03-11T14:00",
+                HttpMethod.PUT,
+                request,
+                ErrorResponse.class);
+
+        assertThat(response.getBody()).isEqualTo(
+                ErrorResponse.builder()
+                        .status(400)
+                        .userMessage("The existing court hearing '-209' must be in a scheduled state to reschedule.")
+                        .developerMessage("The existing court hearing '-209' must be in a scheduled state to reschedule.")
+                        .build());
+    }
+}
```

Comment:
Maybe add a test to create a court hearing then amend it (and do you have a cancel now? If so cancel it) Nice to see full CRUD in a single test

Reply:
No there isn't any ability to cancel a court hearing.  There is for scheduled movements but that is a different entity in NOMIS e.g. prison to prison move.  You would think the same for prison to court but alas no.